src/models/view/nav-tabs.ts

```diff
@@ -56,9 +56,16 @@ export const NavTabSectionModel =
     showStarsForUser(user: UserModelType) {
       return user.type && (self.showStars.indexOf(user.type) !== -1);
     },
-    showDeleteForUser(user: UserModelType) {
-      // allow teachers to delete supports
-      return (user.type === "teacher") && (self.type === ENavTabSectionType.kTeacherSupports);
+    showDeleteForUser(user: UserModelType, userDocument?: any) { //DocumentModelType is a circular logic?
+      const userOwnsDocument = (userDocument && (user.id === userDocument || user.id === userDocument.uid));
+      // allow users to delete published document
+      const deletableTypes = [ENavTabSectionType.kPublishedPersonalDocuments,
+        ENavTabSectionType.kPublishedProblemDocuments,
+        ENavTabSectionType.kPublishedLearningLogs,
+        ENavTabSectionType.kTeacherSupports,
+        ENavTabSectionType.kCurricularSupports,
```

Comment:
Can we really delete curricular supports? I thought the lack of a uid prevented that. If so, we should remove it from this list.